src/internal/parser.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  assertMapDocumentNode,
+  assertProfileDocumentNode,
+  EXTENSIONS,
+  MapDocumentNode,
+  ProfileDocumentNode,
+} from '@superfaceai/ast';
+import { parseMap, parseProfile, Source } from '@superfaceai/parser';
+import { createHash } from 'crypto';
+import { promises as fsp } from 'fs';
+import { join as joinPath } from 'path';
+
+import { Config } from '../config';
+
+export class Parser {
+  private static mapCache: Record<string, MapDocumentNode> = {};
+  private static profileCache: Record<string, ProfileDocumentNode> = {};
+
+  static async parseMap(
+    input: string,
+    fileName: string,
+    info: {
+      profileName: string;
+      providerName: string;
+      scope?: string;
+    }
+  ): Promise<MapDocumentNode> {
```

Comment:
I don't quite understand why this requires the source `input` - ideally it should be able to cache based on the documentId so that you don't have to download the source when you want to check if a map is in the cache.

Reply:
because it caches based on content, for local development